server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/InternalBinaryRange.java

```diff
@@ -70,7 +70,7 @@ private static String generateKey(BytesRef from, BytesRef to, DocValueFormat for
         }
 
         private static Bucket createFromStream(StreamInput in, DocValueFormat format, boolean keyed) throws IOException {
-            String key = in.getVersion().onOrAfter(Version.V_7_17_0) ? in.readOptionalString() : in.readString();
+            String key = in.getVersion().onOrAfter(Version.V_7_17_1) ? in.readOptionalString() : in.readString();
```

Comment:
I think there should be a corresponding change on line 84?

Reply:
Agreed. Let me block this from merging. Make that change. And make sure we have an integration test that would have noticed.

Reply:
We have a test that would have caught it! And it didn't fail for me locally until I ran it one more time! And jenkins isn't running it by default. But I'll add a tag.